packages/legend-studio/src/models/protocols/pure/v1/transformation/pureGraph/from/V1_MappingTransformer.ts

```diff
@@ -464,7 +468,11 @@ const transformRelationalPropertyMapping = (
           [SOURCE_INFORMATION_KEY],
         ),
   ) as V1_RawRelationalOperationElement;
-  propertyMapping.source = undefined; // @MARKER: GRAMMAR ROUNDTRIP --- omit this information during protocol transformation as it can be interpreted while building the graph
+  // Note: isTransformingSourceId is required to have the source information persisted in case of relationalPropertyMappings
```

Comment:
Change this whole comment block to something like:

// NOTE: `isTransformingSourceId` is needed for the roundtrip of association relational property mapping.